test/verify/check-metrics

```diff
@@ -442,6 +444,67 @@ class TestCurrentMetrics(MachineCase):
         # this settles down slowly, don't wait for becoming really quiet
         with b.wait_timeout(180):
             b.wait(lambda: float(b.text("#load-avg").split()[2].rstrip(',')) < 10)
+            b.wait_in_text(".metrics-history", "spike")
+
+        # Check logs for given CPU spike
+        (hour, minute) = self.browser.call_js_func("""(function () {
+            let minute = -1;
+            let hour = -1;
+            const hours = document.getElementsByClassName("metrics-hour");
+            Array.from(hours).some(h => {
+                const minutes = h.getElementsByClassName("metrics-minute");
+                return Array.from(minutes).some(m => {
+                    if (Array.from(m.getElementsByClassName("type")).length) {
+                        minute = m.getAttribute("data-minute");
+                        hour = h.id.split("-")[2];
+                        return true;
+                    }
+                    return false;
+                });
+            });
+
+            return [hour, minute];
+        })""")
```

Comment:
I'm afraid I have some trouble to understand what this does. It seems to search for the first available non-null data and returns its hour/minute. But how does that relate to a "CPU spike", and where/how is that being "given"?

Couldn't that be done in any simpler way? Like a CSS `:first-child` or `:first-of-type` etc, or a `querySelector(..)[0]`?